src/pretix/static/pretixpresale/js/walletdetection.js

```diff
@@ -0,0 +1,120 @@
+            baseCardPaymentMethod
+        );
+
+        return $.ajax({
+            url: 'https://pay.google.com/gp/p/js/pay.js',
+            dataType: 'script',
+            success: function () {
+                const paymentsClient = new google.payments.api.PaymentsClient({environment: 'TEST'});
+                const isReadyToPayRequest = Object.assign({}, baseRequest);
+                isReadyToPayRequest.allowedPaymentMethods = [baseCardPaymentMethod];
+
+                paymentsClient.isReadyToPay(isReadyToPayRequest)
+                    .then(function (response) {
+                        if (response.result) {
+                            return true;
+                        }
+                    })
+                    .catch(function (err) {
+                        return false;
+                    });
+            },
+        });
+    },
+    name_map: {
+        applepay: gettext('Apple Pay'),
+        googlepay: gettext('Google Pay'),
+    }
+}
+
+$(function () {
+    let requestedWallets = Array();
+    let paymentMethods = $('[data-wallets][data-wallets!=""]');
+
+    paymentMethods.each(function () {
+        let $s = $(this);
+        requestedWallets = requestedWallets.concat($s.data("wallets").split("|"));
+    })
+
+    // Filtering out any doubles
+    requestedWallets = new Set(requestedWallets);
+
+    // Perform the actual check *only* on the requested wallets, if they are supported by the browser
+    let availableWallets = Array();
+    requestedWallets.forEach(function (it) {
+        if (walletdetection[it]()) {
+            availableWallets.push(it);
+        }
+    });
+
+    paymentMethods.each(function () {
+        let $s = $(this);
+        let wallets = Array();
+
+        // Run the translation on the available wallet strings before pushing them out.
+        $s.data("wallets").split("|").forEach(function (it) {
+            if (availableWallets.includes(it)) {
+                wallets.push(gettext(walletdetection.name_map[it]));
+            }
+        })
+
+        // In case there is no wallets available, we do not want to flicker the screen
+        if (wallets.length === 0) {
+            return;
+        }
+
+        // The first selector is used on the regular payment-step of the checkout flow
+        // The second selector is used for the payment method change view.
+        // In the long run, the layout of both pages should be adjusted to be one.
+        let textselector = $s.next('label').find('strong');
+        let textselector2 = $s.next("strong");
+        textselector.fadeOut(300, function () {
+            wallets.unshift(textselector.text());
+            textselector.text(wallets.join(", "));
+            textselector.fadeIn(300);
+        });
+        textselector2.fadeOut(300, function () {
+            wallets.unshift(textselector2.text());
+            textselector2.text(wallets.join(", "));
+            textselector2.fadeIn(300);
+        });
```

Comment:
As you run the same code on both elements, you could just combine both selectors:

```suggestion
        $('+ label strong, + strong').fadeOut(300, function () {
            $this = $(this);
            wallets.unshift($this.text());
            $this.text(wallets.join(", "));
            $this.fadeIn(300);
        });
```